CIScripts/Tests/AgentServiceTests.ps1

```diff
@@ -0,0 +1,255 @@
+            return $(Get-Service "ContrailAgent" -ErrorAction SilentlyContinue)
+        }
+        if (!$Service) {
+            throw "Agent service is not registered. EXPECTED: Agent service registered"
+        }
+        if ($Service.Status -eq "Running") {
+            throw "Agent service is running. EXPECTED: Agent service stopped"
+        }
+    }
+
+    #
+    # Tests definitions
+    #
+
+    function Test-AgentServiceIsRegisteredAfterInstall {
+        Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session,
+               [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration)
+
+        Write-Host "===> Running: Test-AgentServiceIsRegisteredAfterInstall"
+
+        Write-Host "======> Given clean environment"
+        Clear-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+        Initialize-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+        Uninstall-Agent -Session $Session
+        Assert-IsAgentServiceUnregistered -Session $Session
+
+        Write-Host "======> When Agent is installed via MSI"
+        Install-Agent -Session $Session
+
+        Write-Host "======> Then Agent is registered as a service"
+        Assert-IsAgentServiceRegistered -Session $Session
+
+        Write-Host "===> PASSED: Test-AgentServiceIsRegisteredAfterInstall"
+    }
+
+    function Test-AgentServiceIsDisabledAfterInstall {
+        Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session,
+               [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration)
+
+        Write-Host "===> Running: Test-AgentServiceIsDisabledAfterInstall"
+
+        Write-Host "======> Given clean environment"
+        Clear-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+        Initialize-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+        Uninstall-Agent -Session $Session
+        Assert-IsAgentServiceUnregistered -Session $Session
+
+        Write-Host "======> When Agent is installed via MSI"
+        Install-Agent -Session $Session
+
+        Write-Host "======> Then Agent is disabled"
+        Assert-IsAgentServiceDisabled -Session $Session
+
+        Write-Host "===> PASSED: Test-AgentServiceIsDisabledAfterInstall"
+    }
+
+    function Test-AgentServiceIsUnregisteredAfterUninstall {
+        Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session,
+               [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration)
+
+        Write-Host "===> Running: Test-AgentServiceIsUnregisteredAfterUninstall"
+
+        Write-Host "======> Given Agent is installed"
+        Clear-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+        Initialize-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+        Install-Agent -Session $Session
```

Comment:
Do we need to install agent here?

Reply:
If we don't want to rely on sequence of tests and we want to use them in different order someday we have to make sure that Agent is installed. In my opinion this test is constructed to be able to run it independently.